server/middleware/AuthenticationMiddleware.js

```diff
@@ -2,25 +2,35 @@ const db = require("../models");
 const { HandledError } = require("./ErrorHandlingMiddleware");
 
 async function authMiddleware(req, res, next) {
-    try {
-      const utorid = req.headers.utorid;
-      const http_mail = req.headers.http_mail;
+  try {
+    let utorid = "";
+    let http_mail = "";
 
-      let user = await db.User.findByPk(utorid);
+    if (process.env?.ENV === "dev") {
+      // Replace shibboleth headers
+      utorid = "demo_utorid";
+      http_mail = "demo@demo.com";
```

Comment:
why not use Requestly to append headers in the browser?

Reply:
There is a reason why we have separate prod and dev environments, why not use it?

Reply:
If I want to use multiple test users, I can set up multiple middleware functions in Requestly and toggle them. This way I have a layer of abstraction over the code.

With node middleware I have to alter the code. This can introduce bugs and increases cognitive load, as you have to think about which file to alter, etc.

Reply:
Fair point, will change it